azcredentials/internal/maputil/maputil.go

```diff
@@ -0,0 +1,87 @@
+package maputil
+
+import "fmt"
+
+func GetMap(obj map[string]interface{}, key string) (map[string]interface{}, error) {
```

Comment:
it would be better to define types for the different structs rather than having to maintain this package and use untyped maps. It's more prone to errors.